lib/rules/no-array-destructured-return.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview Disallow array destructuring of return values.
+ * @author Vitor Balocco
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow array destructuring of return values",
+            category: "Best Practices",
+            recommended: false
+        },
+
+        schema: []
+    },
+
+    create(context) {
+
+        /**
+         * Reports an error on the given node.
+         * @param   {ASTNode} node The node to report on.
+         * @returns {void}
+         */
+        function report(node) {
+            context.report({
+                node,
+                message: "Array destructuring of return values is not allowed, use object destructuring instead."
```

Comment:
@not-an-aardvark honestly I think you are right, i'd personally prefer a rule that does what you mentioned. But the rule as is currently implemented in this PR is how it exists in JSCS so we have to consider what this would mean for JSCS integration.